test/backend/datastores.athena.spec.js

```diff
@@ -0,0 +1,217 @@
+import {assert} from 'chai';
+
+import {connect} from '../../backend/persistent/datastores/athena';
+
+/* eslint-disable no-invalid-this */
+describe('Athena Unit Tests:', function () {
+    it('Invalid connection strings because incorrect credentials', function(done) {
+        const conn = {
+            region: 'us-east-1',
+            accessKey: 'XXXXXXXX',
+            secretKey: 'XXXXXAAAA',
+            database: 'PLOT.LY-TEST',
+            sqlStatement: 'SELECT * FROM TEST_TABLE LIMIT 100',
+            s3Outputlocation: 's3://aws-athena-query-results-11111111-us-east-1/',
+            queryTimeout: 10000
+        };
+        connect(conn).then(connection => {
+           done(`Should NOT have obtained a valid connection ${connection}`);
+        }).catch(err => {
+            assert.isDefined(err, 'Connection failed due to incorrect connection string');
+            done();
+        });
+    });
+
+    it('connect fails missing region', function() {
+        const conn = {
+            accessKey: 'XXXXXXXX',
+            secretKey: 'XXXXXAAAA',
+            database: 'PLOT.LY-TEST',
+            sqlStatement: 'SELECT * FROM TEST_TABLE LIMIT 100',
+            s3Outputlocation: 's3://aws-athena-query-results-11111111-us-east-1/',
+            queryTimeout: 10000
+        };
+
+        return connect(conn).then(connection => {
+            assert.isUnDefined(connection, 'Connection should not have been defined');
+        }).catch(err => {
```

Comment:
We don't want to catch errors. Let mocha handle them. Imagine for example that the assert in line 36 and throws an exception, then it'd be caught here and silenced.